### ISSUE TITLE
False positive for "Use `.Should().HaveSameCount()`"

### ISSUE THREAD

**Issue by FantasyTeddy** [fluentassertions/fluentassertions.analyzers#300]
### Description

The analysis rule [FAA0001](https://fluentassertions.com/tips/#Collections-17): Use `.Should().HaveSameCount()` is triggered for inputs that are actually combined values:

```csharp
// Arrange
var expectedFirstHalf = new List<int> {0};
var expectedSecondHalf = new List<int> {1};

// Act
var actual = new List<int> {10, 11};

// Assert
actual.Should().HaveCount(expectedFirstHalf.Count() + expectedSecondHalf.Count());
```

### Expected behavior:

Since the input (`expectedFirstHalf.Count() + expectedSecondHalf.Count()`) is actually a calculated value, the analysis rule should not be triggered.

### Actual behavior:

The rule is triggered and the code fix crashes.

### Versions

* Fluent Assertions Analyzers version: **0.29.1**
* .NET Runtime/Target: **.NET 8**

**Comment by Meir017** [comment]
@FantasyTeddy Thanks for reporting this!
